### ISSUE TITLE
TypeError: Cannot read property '0' of undefined

### ISSUE THREAD

**Issue by kaidjohnson** [svenschoenung/jshint-stylish-summary#1]
Thanks for this module! Very useful.

We're running into a compatibility issue with gulp-jscs-stylish.

Here's what we're using:

```
jshint 2.9.1
jshint-stylish 2.1.0
jscs 2.11.0
gulp-jscs 3.0.2
gulp-jscs-stylish 1.3.0
gulp-jshint 2.0.0
gulp 3.9.1

node 5.10.0
npm 3.8.3
```

Our linting task is defined like this:

```
gulp.task('lint', () => 
  gulp.src(['app/**/*.js'])
    .pipe(jshint())
    .pipe(jscs())
    .pipe(jscsStylish.combineWithHintResults())
    .pipe(jshint.reporter('jshint-stylish'))
    .pipe(jshintSummary.collect())
    .on('end', jshintSummary.summarize())
);
```

gulp lint then results in:

```
stream.js:74
      throw er; // Unhandled stream error in pipe.
      ^

TypeError: Cannot read property '0' of undefined
  at /path/to/node_modules/jshint-stylish-summary/index.js:167:54
```

It works as expected if I remove the line

```
.pipe(jscsStylish.combineWithHintResults())
```

However, it also works as expected if I leave that line in and I replace `/path/to/node_modules/jshint-stylish-summary/index.js:167` with

```
stats[opts.stat].config.push(file.jshint.results[0]);
```

It seems that the `jscsStylish.combinWithHintResults()` breaks/removes the `file.jshint.data[0]`, but even without jscsStylish using `file.jshint.results[0]` seems to work as expected.

**Comment by svenschoenung** [comment]
Thanks for the report. It has opened a whole can of worms ...

First off there were two issues with your example code:

```
 .pipe(jshint.reporter('jshint-stylish-summary'))
 .pipe(jshintSummary.collection())
```

The second line should be `jshintSummary.collect()`. The first line is unnecessary, since jshint-stylish-summary doesn't support the `jshint.reporter()` function as mentioned in the [docs](https://github.com/svenschoenung/jshint-stylish-summary/#with-gulp-jshint).

After fixing these issues I can reproduce the problem when there is a file that
- has code style problems detected by gulp-jscs
- has no problems detected by gulp-jshint

The reason is that `combineWithHintResults()` [only adds a `.results` property](https://github.com/gonsfx/gulp-jscs-stylish/blob/5bd25e0bf96452532c5ecd854397f7f59368eaac/index.js#L44), but not the `.data` property that gulp-jshint adds. 

This should be an easy fix since the same case already has to be handled when linting is successful. (Your fix works, but I'd prefer to make this consistent.)

However this is only the tip of the iceberg.

If you have a file with problems detected by both gulp-jscs and gulp-jshint you get the following nonsensical output:

![nonsensical-summary](https://cloud.githubusercontent.com/assets/10583730/14471635/c42834de-00ef-11e6-9ca1-9bdebad44fd0.png)

This shows two files as having warnings even though only one file was checked. The reason is that  `combineWithHintResults()` uses absolute paths while gulp-jshint uses project-relative paths. This also causes the output of jshint-stylish to report the same file three times instead of once.

I'll probably fix the `property '0' of undefined` issue tomorrow and then try to get the other stuff fixed in gulp-jscs-stylish itself.

**Comment by svenschoenung** [comment]
Fixed in v0.1.2.

Moved the file count issue to #3

**Comment by kaidjohnson** [comment]
Looks good here.

Sorry for the confusion with collection() and the jshint.reporter call. I've edited my comment to ensure it's not misleading for others.

Also, apologies about the tabs in the PR. May want to consider adding an .editorconfig with configurations that support your jscs rules.

Thanks!
